server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -160,12 +172,12 @@ private void getMultipleReposSnapshotInfo(
                 size,
                 order,
                 groupedActionListener.delegateResponse((groupedListener, e) -> {
-                    if (e instanceof ElasticsearchException) {
-                        groupedListener.onResponse(GetSnapshotsResponse.Response.error(repoName, (ElasticsearchException) e));
+                    if (isMultiRepoRequest && e instanceof ElasticsearchException) {
```

Comment:
I think this way of error handling is new compared to elsewhere. I wonder if you considered one of following options:

* Respond with a failure if all repos failed (which would be the same as here in the single repo case). This is sort of similar to search.
* Breakingly never respond failures, expect users to look into the `failures` map.
* Have an option to fail the request or send back failures in response.

Do we have other similar cases in ES we can lean on in order to decide?

Reply:
> Respond with a failure if all repos failed (which would be the same as here in the single repo case). This is sort of similar to search.

That seems more complicated to work with than having the failures map? My thinking was to have the same behavior for all requests that target multiple repos. I'm somewhat indifferent here, making it look like search is nice, but in isolation I like this approach better I think.

> Breakingly never respond failures, expect users to look into the failures map.

This is quite annoying to users perhaps (note that in 7.x we also have to think about the transport client behavior that would be changed by this). It may also be a troubling breaking change for Cloud? I'd rather not do this I think.

> Have an option to fail the request or send back failures in response.

I don't think we have that kind of behavior anywhere else? Not sure it's worth the complexity for a very small number of use cases?

> Do we have other similar cases in ES we can lean on in order to decide?

None come to mind for me that are similar enough but that doesn't mean there aren't any. I think I was in large part driven by not breaking the transport client API for users and cloud orchestration. In both cases changing the behavior for a single repo request seems risky?

Reply:
We discussed this on another channel with following conclusion:

1. We will merge this PR as is.
2. We will follow-up with a PR to add the option to fail-fast or not, regardless of multi/single repo in regular breaking change style. The default in 7 will be fail fast, default in 8 will be always put it in array. Flag to be removed in 9.